library/Zend/Di/InstanceManager.php

```diff
@@ -502,7 +502,12 @@ protected function createHashForValues($classOrAlias, $paramValues)
                     $hashValue .= $param . '|';
                     break;
                 case 'array':
-                    $hashValue .= 'Array|';
+                    try {
+                        $hashValue .= serialize($param) . '|';
+                    } catch (\Exception $ex) {
```

Comment:
@weierophinney 
really?
```php
serialize(array(function(){}));
```
I did it and I got below. 
```
Exception: Serialization of 'Closure' is not allowed in Command line code on line 1
```

Can ErrorHandler catch this ?